apps/server-asset-sg/src/features/assets/asset.policy.ts

```diff
@@ -0,0 +1,25 @@
+import { Policy } from '@/core/policy';
+import { Asset, AssetData } from '@/features/assets/asset.model';
+import { getRoleIndex, Role } from '@/features/workgroups/workgroup.model';
+
+export class AssetPolicy extends Policy<Asset> {
+  canShow(value: Asset): boolean {
+    // A user can see all assets in all workgroups that they are assigned to.
+    return this.hasWorkgroup(value.workgroupId);
+  }
+
+  canCreate(): boolean {
+    // A user can create assets for workgroups for which they are an Editor.
+    return null != this.user.workgroups.find((group) => getRoleIndex(group.role) >= getRoleIndex(Role.Editor));
```

Comment:
Dont we need to check to which workgroup the user wants to add the asset? This simply checks whether a user is editor in any of their workgroups right?

Reply:
Yes, this is done in the `AssetEditController.create` method. The policy does not have access to the request body, which is why we can only check if the user has the ability to create assets in general. This has the downside of requiring an extra validation of the request body. On the other hand, due to this, we can reuse the policy in other places were there is no body (e.g. in the frontend form, or in other controllers).